testhelpers/testhelpers.go

```diff
@@ -100,6 +103,13 @@ func AssertNil(t *testing.T, actual interface{}) {
 	}
 }
 
+func AssertNotNil(t *testing.T, actual interface{}) {
```

Comment:
Maybe it's only me but I'm a bit confused with the name of the functions `AssertNil` and `AssertNotNil`.
`AssertNil` means that we expect the input to be `nil` and we throw "fatal" if it's `not nil`.
`AssertNotNil` means that we expect the input to be `not nil` and we throw "fatal" if it's `nil`.
What do you think about changing the names of these functions to `FatalOnNil` and `FatalOnNotNil`? But then you need to replace every line that is currently using `AssertNil` to `FatalOnNotNil`, and similarly to `AssertNotNil`.
Another idea is to change the names to be `ExpectNil` and `ExpectNotNil`.
Again, maybe it's only me, so if you think it's clear the way it is right now, I'll deal with my confusion :)

Reply:
It looks like we already have `AssertError()` - how does that sound?

Reply:
As far as I see, `AssertError` is a bit different because it checks that we got an error and that it contains a specific string while in `AssertNil` and `AssertNotNil` we only check whether we got an error or not.
Moreover, I think it's good to have these two functions (maybe `AssertError` can use one of them) but I would only change their names.

Reply:
Given the precedent of `AssertNil` (and its current widespread usage) I'm going to leave `AssertNotNil` for now, but we can revisit if it becomes problematic.